truffle/src/main/java/org/jruby/truffle/core/exception/ExceptionNodes.java

```diff
@@ -133,9 +133,23 @@ public AllocateNode(RubyContext context, SourceSection sourceSection) {
             allocateObjectNode = AllocateObjectNodeGen.create(context, sourceSection, null, null);
         }
 
-        @Specialization
+        @Specialization(guards = "!isNameError(rubyClass)")
         public DynamicObject allocate(DynamicObject rubyClass) {
-            return allocateObjectNode.allocate(rubyClass, nil(), null, nil(), null);
+            return allocateObjectNode.allocate(rubyClass, nil(), null, nil());
+        }
+
+        @Specialization(guards = "isNameError(rubyClass)")
+        public DynamicObject allocateNameError(DynamicObject rubyClass) {
+            return allocateObjectNode.allocate(rubyClass, nil(), null, null, nil());
+        }
+
+        public static boolean isNameError(DynamicObject rubyClass){
+           for(DynamicObject ancestor : Layouts.MODULE.getFields(rubyClass).ancestors()){
+                  if(Layouts.MODULE.getFields(ancestor).getName().equals("NameError")){
```

Comment:
Maybe we should have a checkstyle for simple rules.
I have use Eclipse's formatter too but it needs to be convenient for other IDEs as well.